### ISSUE TITLE
`sharding_constraint_tensor` fails when sharding on mesh axis with size 1

### ISSUE THREAD

**Issue by btsoiTT** [tenstorrent/tt-xla#3285]
### Describe the bug

`sharding_constraint_tensor` fails with an assertion error when attempting to shard on a mesh axis with size 1.
The root cause is `_partition_spec_to_sdy_sharding()` doesn't check for the mesh axes size and always includes the axes from the partition spec. This is found during investigation for https://github.com/tenstorrent/tt-xla/issues/2975.

### Steps to reproduce the issue

```
import os
import torch
import torch_xla
import torch_xla.distributed.spmd as xs
import torch_xla.runtime as xr

from tt_torch.sharding import sharding_constraint_tensor

os.environ["CONVERT_SHLO_TO_SHARDY"] = "1"

xr.set_device_type("TT")
xr.use_spmd()
num_devices = xr.global_runtime_device_count()

mesh_shape = (1, num_devices)  # (1, 8) - batch=1, model=8
device_ids = list(range(num_devices))
mesh = xs.Mesh(device_ids, mesh_shape, ("batch", "model"))

a = torch.ones(128, 128).to("xla")
b = torch.ones(128, 128).to("xla")

xs.mark_sharding(a, mesh, ("batch", None))
xs.mark_sharding(b, mesh, (None, "model"))

c = a @ b

# Sharding on both "batch" (size=1) and "model" (size=8)
c = sharding_constraint_tensor(c, mesh, ("batch", "model"))

print(c)
```

We get the error
```
python: /opt/ttmlir-toolchain/src/shardy/shardy/dialect/sdy/transforms/export/reshard_to_collectives.cc:394: Value mlir::sdy::(anonymous namespace)::CollectiveInserter::insert(): Assertion `isDone()' failed.
```

### Expected behavior

Sharding on axis with size 1 should just be replication and the program should not crash. It should behave the same as `mark_sharding`, where it is represented as `{}` in the MLIR graph from the start.
